rest_framework/static/rest_framework/js/ajax-form.js

```diff
@@ -37,6 +37,19 @@ function doAjaxSubmit(e) {
 
   if (contentType) {
     data = form.find('[data-override="content"]').val() || ''
+
+    if (contentType === 'multipart/form-data') {
+      // We need to add a boundary parameter to the header
+      // We assume the first valid-looking boundary line in the body is correct
+      // regex is from RFC 2046 appendix A
+      var re = /^--([0-9A-Z'()+_,-./:=?]{1,70})[ \f\t\v\u00a0\u1680\u180e\u2000-\u200a\u2028\u2029\u202f\u205f\u3000\ufeff]*$/im;
```

Comment:
The boundary grouping isn't quite correct against the spec, as it *can* include `" "`, it just can't do so at the end.

> The only mandatory global parameter for the "multipart" media type is
> the boundary parameter, which consists of 1 to 70 characters from a
> set of characters known to be very robust through mail gateways, and
> NOT ending with white space.

`boundary := 0*69<bchars> bcharsnospace`